controllers/machinedeployment_sync.go

```diff
@@ -143,12 +144,26 @@ func (r *MachineDeploymentReconciler) getNewMachineSet(ctx context.Context, d *c
 		return nil, err
 	}
 	machineTemplateSpecHash := fmt.Sprintf("%d", hash)
+
+	// If reconciliationTime is after rolloutAfter then a rollout should happen or has already happened.
+	// To trigger it we include the rolloutAfter hash into the MachineSet name so the first time that
+	// reconciliationTime is after rolloutAfter then a new MachineSet is created and the rollout is orchestrated as usual.
+	var rolloutAfterHash uint32
+	if r.reconciliationTime.After(d.Spec.RolloutAfter.Time) {
+		hasher := fnv.New32a()
+		if err := mdutil.SpewHashObject(hasher, d.Spec.RolloutAfter.Time); err != nil {
+			return nil, err
+		}
+		rolloutAfterHash = hasher.Sum32()
+	}
+
+	machineDeploymentHash := fmt.Sprintf("%s%d", machineTemplateSpecHash, rolloutAfterHash)
```

Comment:
If `rolloutAfterHash` is 0, this will still extend the `machineTemplateSpecHash` right? That means that when the deployment controller is upgraded, the hash will change for all existing deployments, causing rollouts of all Machines everywhere even when it wasn't needed.

Should we gate this, eg.
```suggestion
	machineDeploymentHash := machineTemplateSpechHash
	if rolloutAfterHash > 0 {
		machineDeploymentHash = fmt.Sprintf("%s%d", machineTemplateSpecHash, rolloutAfterHash)
	}
```